pkg/test/framework/components/policybackend/policybackend.go

```diff
@@ -308,7 +311,15 @@ func (p *policyBackend) ExpectReport(t testing.TB, expected ...proto.Message) {
 func (p *policyBackend) ExpectReportJSON(t testing.TB, expected ...string) {
 	t.Helper()
 
-	_, err := util.Retry(util.DefaultRetryTimeout, util.DefaultRetryWait, func() (interface{}, bool, error) {
+	var err error
+	for i, e := range expected {
+		expected[i], err = p.env.Evaluate(e)
+		if err != nil {
+			t.Fatalf("template evaluation failed: %v", err)
+		}
+	}
+
+	_, err = util.Retry(util.DefaultRetryTimeout, util.DefaultRetryWait, func() (interface{}, bool, error) {
```

Comment:
btw, do you find it strange that the function that retry accepts returns 3 things? Ideally it should just return an error.